nightingale_navigation/src/room_runner.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python3
+import rospy
+from nightingale_msgs.msg import (
+    RoomRunnerAction,
+    RoomRunnerGoal,
+    RoomRunnerResult,
+    RoomRunnerFeedback,
+)
+from actionlib import SimpleActionServer, SimpleActionClient
+from move_base_msgs.msg import MoveBaseAction, MoveBaseGoal
+from geometry_msgs.msg import PoseStamped
+import tf2_ros
+import tf_conversions
+import os.path
+from actionlib_msgs.msg import GoalStatus
+from geometry_msgs.msg import PoseWithCovarianceStamped
+from enum import Enum
+import numpy as np
+
+
+class RoomRunnerNode(object):
+    class ErrorCode(Enum):
+        SUCCESS = 0
+        NAV_TO_ROOM_FAILED = 1
+        NAV_TO_BEDSIDE_FAILED = 2
+        PREEMPTED_BY_CALLER = 3
+
+    def __init__(self):
+        self.logger_name = "room_runner"
+        rospy.loginfo(
+            "Starting the Nightingale Navigation Room Runner Node!",
+            logger_name=self.logger_name,
+        )
+
+        self.tf_buffer = tf2_ros.Buffer()
+        self.tf_listener = tf2_ros.TransformListener(self.tf_buffer)
+
+        self.save_pose = rospy.get_param("/room_runner/save_pose", False)
+        self.pose_filename = rospy.get_param("/room_runner/pose_filename", "")
+        self.pose_write_period = rospy.get_param("/room_runner/pose_write_period", 60)
```

Comment:
thats in seconds. though we write the pose on rospy.shutdown, we also write at some period just in case the node collapses suddenly or computer looses power or something unexpected. This is already pretty unlikely. The only reason we write the pose to disk is so that we have an initial guess when we wake back up. we could make it more frequent maybe 10 sec

Reply:
If the pose is written on shutdown that's good